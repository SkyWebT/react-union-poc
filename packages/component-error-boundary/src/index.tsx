import React from 'react';

class ErrorBoundary extends React.Component<{}, { hasError: boolean }> {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error) {
    // Update state so the next render will show the fallback UI.
    return { hasError: true };
  }

  componentDidCatch(error, errorInfo) {
    // You can also log the error to an error reporting service
    console.log(error, errorInfo);
  }

  render() {
    if (this.state.hasError) {
      // You can render any custom fallback UI
      return (
        <div style={{ backgroundColor: 'red' }}>
          <h2>Something went wrong</h2>
          <p>Our engineer team is aware of this issue. </p>
        </div>
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
