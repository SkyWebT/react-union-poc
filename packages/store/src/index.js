export { create ,actions} from './store';
