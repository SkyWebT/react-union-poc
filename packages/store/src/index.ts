export { create, actions, AppState } from './store';
