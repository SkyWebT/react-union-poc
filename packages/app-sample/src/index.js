import React from 'react';
import { Provider } from 'react-redux';
import { justRender } from 'react-union';
import { create } from '@skytv/store';

import Root from './components/Root';

const store = create();

const app = (
	<Provider store={store}>
		<Root />
	</Provider>
);

if ('Liferay' in window) {
	window.Liferay.on('allPortletsReady', function() {
		console.log('allPortletsReady !!!!!!!');
		justRender(app);
	});
} else {
	justRender(app);
}
