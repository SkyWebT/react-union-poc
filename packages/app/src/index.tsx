import React from 'react';
import { Provider } from 'react-redux';
import { justRender } from 'react-union';

import { create } from '@sky/store';

import Root from './Root';

const store = create();

const app = (
  <Provider store={store}>
    <Root />
  </Provider>
);

declare global {
  interface Window {
    Liferay: any;
  }
}
if ('Liferay' in window) {
  window.Liferay.on('allPortletsReady', function() {
    console.log('allPortletsReady !!!!!!!');
    justRender(app);
  });
} else {
  justRender(app);
}
