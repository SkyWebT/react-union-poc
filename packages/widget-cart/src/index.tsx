import React from 'react';
import { connect } from 'react-redux';
import { createSelector, PayloadActionCreator } from 'redux-starter-kit';

import { actions, AppState, T_Product } from '@skytv/store';

const selectSelected = (state: AppState) => Object.values(state.products).filter(p => p.selected);

const selector = createSelector(
  selectSelected,
  products => ({
    total: products.reduce((acc, cur) => {
      acc += cur.price;
      return acc;
    }, 0),
    products,
  })
);

interface Props {
  products: T_Product[];
  total: number;
  toggle: PayloadActionCreator<string>;
}
const Root: React.FC<Props> = ({ products, total, toggle }) => {
  const removeFromCart = (id: string) => {
    toggle(id);
  };
  return (
    <div style={{ border: '1px solid grey', padding: 20, width: '40vw' }}>
      <h2>Cart </h2>
      {products.length > 0 ? (
        products.map(({ id, name, price }) => (
          <div key={id} style={{ display: 'flex', justifyContent: 'space-between' }}>
            <h3>
              {name} - ${price}
            </h3>
            <div>
              <button onClick={() => removeFromCart(id)}>remove</button>
            </div>
          </div>
        ))
      ) : (
        <div> empty </div>
      )}
      <h3>Total: ${total}</h3>
    </div>
  );
};

export default connect(
  selector,
  actions.prodducts
)(Root);
