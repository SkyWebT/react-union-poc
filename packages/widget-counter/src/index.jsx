import { useSelector, useDispatch } from 'react-redux';
import React from 'react';
import { actions } from '@skytv/store';
import style from './style.css';

const Root = () => {
  const { counter } = useSelector(s => s.counter);
  const d = useDispatch();
  const inc = () => {
    d(actions.counter.increment());
  };
  const dec = () => {
    d(actions.counter.decrement());
  };
  return (
    <div className={style.counter}>
      <div>hello typescript counter</div>
      <div>{counter}</div>
      <button onClick={inc}>+++</button>
      <button onClick={dec}>---</button>
    </div>
  );
};

export default Root;
