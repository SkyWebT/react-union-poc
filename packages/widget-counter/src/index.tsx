import React from 'react';
import { connect } from 'react-redux';

import { actions, AppState } from '@skytv/store';

import style from './style.css';

interface Props {
  counter: number;
  increment: () => void;
  decrement: () => void;
}

export const Counter: React.FC<Props> = ({ counter, increment, decrement }) => {
  if(counter==1){
    throw 'err'
  }
  return (
    <div className={style.counter}>
      <div>hello typescript counter</div>
      <div>{counter}</div>

      <button onClick={increment}>+++</button>
      <button onClick={decrement}>---</button>
    </div>
  );
};

export default connect(
  (s: AppState) => ({ counter: s.counter }),
  actions.counter
)(Counter);
