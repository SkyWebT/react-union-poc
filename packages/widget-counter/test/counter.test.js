import React from 'react';
import { Counter } from '../src/index';
import { render } from '@testing-library/react';
describe('<Counter />', () => {
  const requiredProps = {
    counter: 1,
    increment: () => {},
    decrement: () => {},
  };
  it('should match snapshot', () => {
    const { asFragment } = render(<Counter {...requiredProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should show text 1', () => {
    const { getByText } = render(<Counter {...requiredProps} />);
    getByText('1');
  });
});
