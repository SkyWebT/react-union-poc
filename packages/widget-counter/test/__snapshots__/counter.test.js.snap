// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      hello typescript counter
    </div>
    <div>
      1
    </div>
    <button>
      +++
    </button>
    <button>
      ---
    </button>
  </div>
</DocumentFragment>
`;
